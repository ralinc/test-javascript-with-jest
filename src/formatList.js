export default formatList

function formatList(listName, items, key) {
  return `List ${listName}:${
    items.reduce((formatted, item) => {
      return `${formatted}\n  - ${key ? item[key] : item}`
    }, '')
  }`
}
