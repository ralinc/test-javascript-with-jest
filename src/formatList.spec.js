import formatList from './formatList'

test('formats a list', () => {
  const formattedList = formatList(
    'List foo:',
    [
      {name: 'bar'},
      {name: 'baz'},
    ],
    'name'
  )

  // console.log(formattedList);

  expect(formattedList).toMatchSnapshot()
})
