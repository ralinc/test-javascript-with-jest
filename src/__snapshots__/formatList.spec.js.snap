// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats a list 1`] = `
"List List foo::
  - bar
  - baz"
`;
